### PR TITLE
[HD32][InContextLearning] Add dismiss button and onDismiss prop

### PR DESCRIPTION
**Pull request by annmariedoyle-shopify**
Adds an onDismiss prop and a dismiss button that calls onDismiss. 

### WHY are these changes introduced?

More information on these changes can be found [here](https://docs.google.com/document/d/1Jf-6lPq5lygtFsZf9wt0vD5i7fKt6bHhrCHFL8-fjVc/edit).

### WHAT is this pull request doing?

Add a new function prop `onDismiss` to the component and renders a button a close button

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

File: polaris-react/locales/en.json [hunk 1]
```diff
@@ -155,6 +155,9 @@
       "clear": "Clear",
       "clearLabel": "Clear {filterName}"
     },
+    "InContextLearning": {
+      "accessibilityLabel": "Close tutorial"
```
Comment (annmariedoyle-shopify) [inline polaris-react/locales/en.json:159]: I got this content from this doc: https://docs.google.com/document/d/10x1ynHfpZVMDGKEhqIMhXhIT3suC8rdex4Dh1gbAJ3g/edit

I wasn't sure if we wanted to have a prop of the tutorial name passed into the component - at this point it seems unnecessary for just one label 🤷‍♀️  But I can update if we feel that:
1) having the name in this label is important
2) we have a use for the name beyond this label

File: polaris-react/src/components/InContextLearning/InContextLearning.tsx [hunk 2]
```diff
@@ -1,37 +1,56 @@
...
+    console.warn('Next clicked');
+  };
+
+  const handlePrev = () => {
+    setCurrentStep((currentStep) => currentStep - 1);
+    console.warn('Prev clicked');
+  };
+
+  const dismissButton = onDismiss ? (
+    <div>
```
Comment (annmariedoyle-shopify) [inline polaris-react/src/components/InContextLearning/InContextLearning.tsx:34]: I assume styles will be added under a different PR

File: polaris-react/locales/en.json (hunk 1 above)
Comment (kimhawthorne-shopify) [reply polaris-react/locales/en.json:159]: I think this is fine for hack days — we can totally revisit if we want to make this more specific

**Review by rockinrobin714 — APPROVED** [review]
lgtm!

**Review by kimhawthorne-shopify — APPROVED** [review]
Looks good to me!

File: polaris-react/src/components/InContextLearning/InContextLearning.tsx [hunk 3]
```diff
@@ -1,37 +1,58 @@
...
+
+import {useI18n} from '../../utilities/i18n';
+
 import {Step} from './components';
 import {Button} from '../Button';
 
 interface Props {
-    children: React.ReactElement[];
+  children: React.ReactElement[];
+  onDismiss?(): void;
```
Comment (arapehl) [inline polaris-react/src/components/InContextLearning/InContextLearning.tsx]: I read the doc outlining the pattern we're following from elsewhere, but since the design calls for a dismiss "X" button and `esc` behaviour, are sure we want to make this optional?

Comment (annmariedoyle-shopify) [reply polaris-react/src/components/InContextLearning/InContextLearning.tsx]: That's a good question. I was thinking that we would want things to be flexible and not force it to be dismissible but rather allow that as an option.

Comment (annmariedoyle-shopify) [reply polaris-react/src/components/InContextLearning/InContextLearning.tsx]: Looking at other Polaris components, this seems to be a common pattern - to allow some configuration/flexibility within the component.

Comment (arapehl) [reply polaris-react/src/components/InContextLearning/InContextLearning.tsx]: True, but in the case of banners, there are dismissible ones and non-dismissible ones. In our case, I think all we have are dismissible elements. We could check with UX to confirm.

Comment (annmariedoyle-shopify) [reply polaris-react/src/components/InContextLearning/InContextLearning.tsx]: Sure, I will ask in the channel.

Comment (annmariedoyle-shopify) [reply polaris-react/src/components/InContextLearning/InContextLearning.tsx]: Confirmed in the change the the `onDismiss` functionality in the [channel](https://shopify-hack-days.slack.com/archives/C03BFM4D54Y/p1652125473077789?thread_ts=1652125260.072959&cid=C03BFM4D54Y). I think the PR to close using `esc` should be a separate PR to ensure we are keeping out PRs small.

**Review by arapehl — APPROVED** [review]
Code is good. Design decision behind onDismiss can be hammered out even after this PR ships.
